src/waitFor.js

```diff
@@ -24,20 +25,24 @@ function waitForInternal<T>(
   options?: WaitForOptions
 ): Promise<T> {
   const timeout = options?.timeout ?? DEFAULT_TIMEOUT;
-  const interval = options?.interval ?? DEFAULT_INTERVAL;
-  const startTime = Date.now();
+  let interval = options?.interval ?? DEFAULT_INTERVAL;
+
+  if (interval < 1) interval = 1;
+  const maxTries = Math.ceil(timeout / interval);
```

Comment:
maxTries maybe wrong, since `interval` is not guaranted to be exact, for small intervals the delay might be larger due to evenc cycle time. 

> delay Optional
>     The time, in milliseconds (thousandths of a second), the timer should wait before the specified function or code is executed. If this parameter is omitted, a value of 0 is used, meaning execute "immediately", or more accurately, the next event cycle. Note that in either case, the actual delay may be longer than intended; see Reasons for delays longer than specified below.

https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/setTimeout

Not sure if that affects your code logic.

Reply:
It's an interesting point but I don't think it affects things. There is a slight delay when scheduling timeouts and this may lead to a case where a timeout is reached later than it should have been, but this should only happen if the `interval` is 0 or otherwise very low. We could set a minimum `interval`?

Other notes:
- it's a good idea to not rely on `Date` since this can also be mocked
- this approach is what RNTL v6 used
- in the future this can be improved by borrowing an approach from `dom-testing-library`

Reply:
Maybe we should adopt wait-for from DOM Testing Library, what do you think? https://github.com/testing-library/dom-testing-library/blob/master/src/wait-for.js
Interesting thing is that inside they advance fake timers, which we currently miss. There may be some other small edge-cases they handle already. Sounds like a sane thing to do

Reply:
Changed `waitFor` to be based on DOM Testing Library's implementation